src/models.rs

```diff
@@ -29,8 +29,31 @@ pub(crate) struct ConfigurationJson {
     pub segments: Vec<Segment>,
 }
 
+impl ConfigurationJson {
+    /// Parses a ConfigurationJson from a file
+    pub(crate) fn new(filepath: &std::path::Path) -> Result<Self> {
+        let file = std::fs::File::open(filepath).map_err(|_| {
```

Comment:
where does this come from? Is this in anticipation of providing offline default values?

Reply:
ah found it, was previously implemented in `app_configuration_offline.rs` now here.